io/zenoh-transport/src/unicast/universal/link.rs

```diff
@@ -191,8 +191,10 @@ async fn tx_task(
                             stats.inc_tx_bytes(batch.len() as usize);
                         }
 
-                        // Reinsert the batch into the queue
-                        pipeline.refill(batch, priority);
+                        if !batch.is_ephemeral() {
```

Comment:
I would move the check inside the `refill` function for better encapsulation and avoid checking wether a batch is ephemeral every time it is used.